src/NetTopologySuite/Geometries/GeometryFactoryEx.cs

```diff
@@ -205,6 +206,21 @@ public override MultiPolygon CreateMultiPolygon(Polygon[] polygons)
             return new MultiPolygon(polygons, this);
         }
 
+        /// <inheritdoc />
+        public override GeometryFactory WithSRID(int srid)
+        {
+            var clone = base.WithSRID(srid);
+            if (!(clone is GeometryFactoryEx cloneEx))
+            {
+                cloneEx = new GeometryFactoryEx(PrecisionModel, srid, CoordinateSequenceFactory, GeometryServices);
+            }
```

Comment:
actually, let's always do the explicit copy... base version via `NtsGeometryServices` might return a shared instance that we shouldn't be messing with.